app/lib/backend/model_extensions/cinstance.rb

```diff
@@ -50,28 +45,6 @@ def delete_backend_application
         true
       end
 
-      def update_backend_user_key_to_application_id_mapping
-        user_key_was, current_user_key = previous_changes[:user_key]
-
-        if previously_changed?(:user_key) && service.id.present? && user_key_was.present?
-          ThreeScale::Core::Application.delete_id_by_key(service.backend_id, user_key_was)
-        end
-
-        ## save no matter what, even if not changed, it's safe. Required for backend rake task that are
-        ## unaware of previous changes
-        if !service.nil? && user_key.present?
-          ThreeScale::Core::Application.save_id_by_key(service.backend_id, user_key, application_id)
-        end
-
-        true
-      end
-
-      def delete_backend_user_key_to_application_id_mapping
-        return true if !service || !service.id || user_key.blank?
-        ThreeScale::Core::Application.delete_id_by_key(service.backend_id, user_key)
```

Comment:
It looks like we do not pass the `user_key` in

https://github.com/3scale/porta/blob/70cd39602a28c9d451b1aaaa3c4656145cb7466e/app/lib/backend/model_extensions/cinstance.rb#L25

Referring to https://github.com/3scale/porta/blob/70cd39602a28c9d451b1aaaa3c4656145cb7466e/app/lib/backend/model_extensions/cinstance.rb#L46

Reply:
exactly 😄